### PR TITLE
More fields added in SlackUserDetails model. 

### PR DESCRIPTION
**Pull request by juliepromact**
Fields like Deleted or whether user Is_Bot is checked before adding them to database.

### CODE REVIEW

**Comment by chintans** [comment]
Fixed Issue #56

**Review by chintans — CHANGES_REQUESTED** [review]
@juliepromact  essentially added just 1 property. When I say more fields than it means more than 1.

File: Slack.Automation/Promact.Core.Repository/ExternalLoginRepository/OAuthLoginRepository.cs [hunk 1]
```diff
@@ -78,27 +84,29 @@ public OAuthApplication ExternalLoginInformation(string refreshToken)
             var slackUsers = JsonConvert.DeserializeObject<SlackUserResponse>(userDetailsResponse);
             foreach (var user in slackUsers.Members)
             {
-                if (user.Name != StringConstant.SlackBotStringName)
-                {
-                    user.CreatedOn = DateTime.UtcNow;
-                    _slackUserDetails.Insert(user);
-                    _slackUserDetails.Save();
-                }
+                if (!user.Deleted && user.IsBot == false)
```
Comment (chintans) [inline Slack.Automation/Promact.Core.Repository/ExternalLoginRepository/OAuthLoginRepository.cs]: Improve this condition. if you are using a bool property then don't compare it with a bool value.

**Review by chintans — CHANGES_REQUESTED** [review]

File: Slack.Automation/Promact.Core.Repository/ExternalLoginRepository/OAuthLoginRepository.cs (hunk 1 above)
Comment (juliepromact) [reply Slack.Automation/Promact.Core.Repository/ExternalLoginRepository/OAuthLoginRepository.cs]: Improved condition

File: Slack.Automation/Promact.Core.Repository/ExternalLoginRepository/OAuthLoginRepository.cs [hunk 2]
```diff
@@ -78,27 +84,29 @@ public OAuthApplication ExternalLoginInformation(string refreshToken)
             var slackUsers = JsonConvert.DeserializeObject<SlackUserResponse>(userDetailsResponse);
             foreach (var user in slackUsers.Members)
             {
-                if (user.Name != StringConstant.SlackBotStringName)
-                {
-                    user.CreatedOn = DateTime.UtcNow;
-                    _slackUserDetails.Insert(user);
-                    _slackUserDetails.Save();
-                }
+                if (!user.Deleted && !user.IsBot && user.Name != StringConstant.SlackBotStringName)
```
Comment (chintans) [inline Slack.Automation/Promact.Core.Repository/ExternalLoginRepository/OAuthLoginRepository.cs:87]: Why did you add additional condition of name?

**Review by chintans — CHANGES_REQUESTED** [review]

File: Slack.Automation/Promact.Core.Repository/ExternalLoginRepository/OAuthLoginRepository.cs (hunk 2 above)
Comment (juliepromact) [reply Slack.Automation/Promact.Core.Repository/ExternalLoginRepository/OAuthLoginRepository.cs:87]: Sir, otherwise slackbot would be added to SlackUserDetails tables, as it is neither marked(is_bot=false) as a bot by slack nor deleted,the two conditions that we check to be false while adding slack users to database.

**Review by chintans — APPROVED** [review]
